omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,153 @@
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addSelfLink();
+			return description;
+		}
+		
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("identifierType");
+			
+			// links
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			}
+		}
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getidentifierType(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType().getName();
+	}
+	
+	@PropertySetter("identifierType")
+	public static void setidentifierType(IdentifierSource identifierSource, PatientIdentifierType patientIndentifierType) {
+		identifierSource.setIdentifierType(patientIndentifierType);
+		
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		return new SequentialIdentifierGenerator();
+	}
+	
+	@Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+		Object savedIdentifierSource = null;
+		
+		if(postBody.get("sourceType").toString().equals("SequentialIdentifierGenerator")) {
+			
+			String baseCharacterSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+			
+			SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+			
+			String uuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			String description = postBody.get("description").toString();
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setBaseCharacterSet(baseCharacterSet);
+			identifierSource.setFirstIdentifierBase("1");
```

Comment:
Did you intentionally hard code the above "1"?

Reply:
No. The [test data](https://github.com/openmrs/openmrs-module-idgen/blob/master/api/src/test/resources/org/openmrs/module/idgen/include/TestDataForLoadTest.xml) and the [identifierSourceServiceTest file](https://github.com/openmrs/openmrs-module-idgen/blob/master/api/src/test/java/org/openmrs/module/idgen/service/IdentifierSourceServiceTest.java#L140) both use 1 for the first_identifier_base variable.

Reply:
So what happens to those users whose first identifier base is not "1"?

Reply:
True. Fixed. The first identifier base is now coming from the test/rest call.

Reply:
Are the fixes committed?

Reply:
yes